### DISCUSSION TITLE
Custom Exporters for Calendar Events

### DISCUSSION THREAD

**Discussion by sadmachine** [solspace/craft-calendar#265] [Questions & Answers] [answered]
Was hoping for some help regarding exporting calendar events from the solspace/craft-calendar plugin. For standard craft entries, you can define and register custom exporters that will show up in the "Export Type" list like below:
![image](https://github.com/solspace/craft-calendar/assets/9452483/e605ab6d-4660-407b-9f5f-00505adcf630)

However, when I go to export events from the Calendar plugin, those custom exporters do not show up:
![image](https://github.com/solspace/craft-calendar/assets/9452483/ec7ba545-3a7e-4bf5-81b2-721a5f725011)

Is there anyway to define custom exporters for these events? The standard format of the export is not workable and we would like to define a custom export to only grab the information we need in a format that is more workable.

**Comment by seandelaney** [comment, marked as answer]
It should work exactly the same as your People Exporter.

Also see: https://craftcms.com/docs/5.x/extend/element-exporter-types.html

I created:

```
<?php

namespace namespace\foobar;

use craft\base\Element;
use craft\base\ElementExporter;
use craft\elements\db\ElementQueryInterface;

class MyCustomCalendarEventExporter extends ElementExporter
{
    public static function displayName(): string
    {
        return 'My Custom Calendar Event Exporter';
    }

    public function export(ElementQueryInterface $query): mixed
    {
        $results = [];

        foreach ($query->each() as $element) {
            /** @var Element $element */
            $results[] = [
                'Title' => $element->title ?? '',
                'Status' => ucfirst($element->status),
                'URL' => $element->getUrl(),
            ];
        }

        return $results;
    }
}
```

Then in a custom module, I called:

```
Event::on(
   \Solspace\Calendar\Elements\Event::class,
   Element::EVENT_REGISTER_EXPORTERS,
   function(RegisterElementExportersEvent $event) {
      $event->exporters[] = MyCustomCalendarEventExporter::class;
   }
);
```

When I view Events index page under Calendar plugin, I see my custom exporter listed:

![image](https://github.com/solspace/craft-calendar/assets/580403/9ee2321b-ec19-4042-979b-737f37a47c7b)
